### PR TITLE
Added supported by organization type to listbox with options,allowed …

### PR DESCRIPTION
**Pull request by roshni73**
## Addresses:
-  https://github.com/IFRCGo/go-web-app/issues/451

## Changes
- Added supported by organization type to listbox with options,
- allowed the selection of a National Society when Supported by organization type is National Society.

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx [hunk 1]
```diff
@@ -111,16 +114,32 @@ function PrioritizedActionInput(props: Props) {
...
-                value={value?.supported_by}
-                error={error?.supported_by}
+                keySelector={organizationTypeKeySelector}
+                labelSelector={stringValueSelector}
+                value={value?.supported_by_organization_type}
+                error={error?.supported_by_organization_type}
                 readOnly={readOnly}
                 disabled={disabled}
             />
+            {value?.supported_by_organization_type === 3 && (
```
Comment (barshathakuri) [inline src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx]: Please compare with the enum. Let me know if you have any confusion.

**Review by barshathakuri — CHANGES_REQUESTED** [review]

**Review by samshara — APPROVED** [review]

File: src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx [hunk 2]
```diff
@@ -57,9 +62,9 @@ function PrioritizedActionInput(props: Props) {
     } = props;
 
     const { per_workplanstatus } = useGlobalEnums();
+    const { per_supported_by_organization_type } = useGlobalEnums();
```
Comment (barshathakuri) [inline src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx:65]:
```suggestion
    const { per_workplanstatus, per_supported_by_organization_type } = useGlobalEnums();
```

File: src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx [hunk 3]
```diff
@@ -17,10 +18,11 @@ import useGlobalEnums from '#hooks/domain/useGlobalEnums';
 import { type GoApiResponse } from '#utils/restRequest';
 import { stringValueSelector } from '#utils/selectors';
 import { resolveToString } from '#utils/translation';
-
+import { NATIONAL_SOCIETY } from '#utils/constants';
 import { type PartialWorkPlan } from '../schema';
 
 import i18n from './i18n.json';
+
```
Comment (barshathakuri) [inline src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx:25]: Remove extra line

File: src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx (hunk 2 above)
Comment (barshathakuri) [inline src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx]:
```suggestion
    const { per_workplanstatus, per_supported_by_organization_type } = useGlobalEnums();
```

File: src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx (hunk 3 above)
Comment (barshathakuri) [inline src/views/PerWorkPlanForm/PrioritizedActionInput/index.tsx]: Remove extra line

**Review by barshathakuri — CHANGES_REQUESTED** [review]

**Review by barshathakuri — APPROVED** [review]

**Pull request merged**